standard/interfaces.md

```diff
@@ -305,75 +404,184 @@ These rules ensure that any covariant or contravariant usage of the interface re
+- Otherwise, if `T` is a class and a direct or indirect base class declares an implementation of `I.M`, the most derived base class of `T` is the most specific override.
+- Otherwise, if `I2` and `I3` are interfaces implemented by `T` and `I3` derives from `I2` either directly or indirectly, `I3.M` is a more specific implementation than `I2.M`.
+- Otherwise, neither `I2.M` nor `I3.M` are more specific and an error occurs.
+
+> *Example*:
+>
+> <!-- Example: {template:"standalone-lib-without-using", name:"InterfaceMethods4", expectedErrors:["CS8705"]} -->
+> ```csharp
+> interface IA
+> {
+>     void M() { Console.WriteLine("IA.M"); }
+> }
+> 
+> interface IB : IA
+> {
+>     void IA.M() { Console.WriteLine("IB.M"); }
+> }
+> 
+> interface IC: IA
+> {
+>     void IA.M() { Console.WriteLine("IC.M"); }
+> }
+> 
+> abstract class C: IB, IC { } // error: no most specific override for 'IA.M'
+> 
+> abstract class D: IA, IB, IC // OK
+> {
+>     public abstract void M();
+> }
+> ```
+>
+> The most specific override rule ensures that a conflict (i.e., an ambiguity arising from diamond inheritance) is resolved explicitly by the programmer at the point where the conflict arises. *end example*
+
+It is an error if in a class declaration the most specific override of some interface method is an abstract override that was declared in a class or an interface.
+
+> *Example*:
+>
+> <!-- Example: {template:"standalone-lib-without-using", name:"InterfaceMethods5", expectedErrors:["CS0535"]} -->
+> ```csharp
+> interface IF
+> {
+>     void M();
+> }
+> 
+> abstract class F : IF { } // error: 'F' does not implement 'IF.M'
+> ```
+>
+> *end example*
+
+It is an error if a property `P` has a most specific `get` accessor in one type `T1`, and a most specific `set` accessor in another type `T2`.
 
 ### 18.4.6 Interface member access
 
-Interface members are accessed through member access ([§12.8.7](expressions.md#1287-member-access)) and indexer access ([§12.8.12.3](expressions.md#128123-indexer-access)) expressions of the form `I.M` and `I[A]`, where `I` is an interface type, `M` is a method, property, or event of that interface type, and `A` is an indexer argument list.
+Interface members are accessed through member access ([§12.8.7](expressions.md#1287-member-access)) and indexer access ([§12.8.12.3](expressions.md#128123-indexer-access)) expressions of the form `I.M` and `I[A]`, where `I` is an interface type, `M` is a constant, field, method, property, or event of that interface type, and `A` is an indexer argument list. If `M` is non-abstract, the name `M` is visible only in the interface `I` and interfaces derived from `I`.
+
+In a class `D`, with base class `B`, where `B` implements interface `I` and `I` defines a method `M()`:
+
+- The expression `base.M()` is valid only if `B` provides a definition for `M`. The expression `base.M()` is valid only if the most specific implementation (§most-specific-implementation) of `M` is in a class type.
```

Comment:
Two sentences, both starting “The expression `base.M()` is valid only if”. Is this meant to be a conjunction, or was one sentence meant to replace the other?

Reply:
refactored a bit in next commit.